### PR TITLE
Refactor: Popup View and Controller files

### PR DESCRIPTION
**Pull request by chynu**
Added semi-empty (not fully implemented) view and controller classes for popups. `PopupController` controls all popups that may appear while using Devtools. The view of the `PopupController` changes depending on which view is active. There is a base view class called `PopupView`. 

Classes will be implemented in future PR's.

---
This change is [Reviewable](https://reviewable.io/reviews/google/blockly-devtools/68)

### CODE REVIEW

File: src/view/new_block_popup_view.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    /**
+     * Popup contents in HTML format.
+     * @type {string}
+     */
+    this.htmlContents = `
+<div class="bkg"></div>
+<div class="box">
+  <div id="exit">x</div>
+  <header>Create a New Block</header>
```
Comment (AnmAtAnm) [inline src/view/new_block_popup_view.js]: I think the above lines are shared by all three implementations, and the base class `PopupView` will need to know about about them to implement `hide()`.  Consider implementing these container `<div>`s in the superclass.

**Comment by rachel-fenichel** [comment]
Reviewed 9 of 10 files at r1.
Review status: 9 of 10 files reviewed at latest revision, 3 unresolved discussions.

---

*[src/controller/app_controller_2.js, line 88 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/68#-Knv5K-EUr3LyvZvGrYV:-Knv5K-EUr3LyvZvGrYW:bhttl72) ([raw file](https://github.com/google/blockly-devtools/blob/385eb19e697c2e91871f9e89e4489215da1876ee/src/controller/app_controller_2.js#L88)):*
> ```JavaScript
>   /**
>    * Generates popup. Param must be either this.popupController.MODE.PREVIEW,
>    * this.popupController.MODE.NEW_BLOCK, or this.popupController.MODE,NEW_CONFIG.
> ```

period, not comma

---

*[src/view/preview_view.js, line 33 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/68#-Knv61ulgFsJI6tRhHAE:-Knv61ulgFsJI6tRhHAF:b-ju0nm3) ([raw file](https://github.com/google/blockly-devtools/blob/385eb19e697c2e91871f9e89e4489215da1876ee/src/view/preview_view.js#L33)):*
> ```JavaScript
> 'use strict';
> 
> class PreviewView extends PopupView {
> ```

WorkspacePreviewView?  You could be previewing all sorts of things.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-devtools/68)*

**Comment by chynu** [comment]
Review status: 5 of 10 files reviewed at latest revision, 3 unresolved discussions.

---

*[src/controller/app_controller_2.js, line 88 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/68#-Knv5K-EUr3LyvZvGrYV:-KnvFi96CYB-bhZmJ6n3:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/385eb19e697c2e91871f9e89e4489215da1876ee/src/controller/app_controller_2.js#L88)):*
<details><summary><i>Previously, rachel-fenichel (Rachel Fenichel) wrote…</i></summary><blockquote>

period, not comma
</blockquote></details>

Done.

---

*[src/view/new_block_popup_view.js, line 41 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/68#-Knv1MZV4gJ7ZMGVJ6Ec-r1-41:-KnvFjALd-uBtvAzP79k:b-xgri42) ([raw file](https://github.com/google/blockly-devtools/blob/385eb19e697c2e91871f9e89e4489215da1876ee/src/view/new_block_popup_view.js#L41)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

I think the above lines are shared by all three implementations, and the base class `PopupView` will need to know about about them to implement `hide()`.  Consider implementing these container `<div>`s in the superclass.
</blockquote></details>

Ooh, good point. I shifted some code around.

---

*[src/view/preview_view.js, line 33 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/68#-Knv61ulgFsJI6tRhHAE:-KnvARDTRtXz7gJxFN2q:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/385eb19e697c2e91871f9e89e4489215da1876ee/src/view/preview_view.js#L33)):*
<details><summary><i>Previously, rachel-fenichel (Rachel Fenichel) wrote…</i></summary><blockquote>

WorkspacePreviewView?  You could be previewing all sorts of things.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-devtools/68)*

**Comment by AnmAtAnm** [comment]
LGTM

**Pull request merged**